jax/_src/random.py

```diff
@@ -1724,3 +1724,83 @@ def ball(
   g = generalized_normal(k1, p, (*shape, d), dtype)
   e = exponential(k2, shape, dtype)
   return g / (((jnp.abs(g) ** p).sum(-1) + e) ** (1 / p))[..., None]
+
+def vonmises(
+  key: KeyArray,
+  x: float,
+  kappa: float,
+  shape: Shape = (),
+  dtype: DTypeLikeFloat = dtypes.float_
+) -> Array:
+  """Draw samples from a von Mises distribution.
+
+  Reference:
+    Best, D. J., & Fisher, N. I. (1979). Efficient Simulation of the
+    von Mises Distribution. Applied Statistics, 28(2), 152.
+    https://doi.org/10.2307/2346732
+
+  Args:
+    key: a PRNG key used as the random key.
+    x: mean of the distribution.
+    kappa: concentration of the distribution.
+    shape: shape of the output array.
+    dtype: dtype of the output array.
+
+  Returns:
+    theta: A random array of specified shape and dtype."""
+  # Run Checks
+  key, _ = _check_prng_key(key)
+  if not dtypes.issubdtype(dtype, np.floating):
+    raise ValueError(f"dtype argument to `t` must be a float"
+                     f"dtype, got {dtype}.")
+  dtype = dtypes.canonicalize_dtype(dtype)
+  shape = core.canonicalize_shape(shape)
+
+  # Return Distribution
+  if kappa <= 1e-6:
+    return ((jnp.pi * 2) * uniform(key, shape, dtype)).astype(dtype)
+  else:
+    return _vonmises(key, x, kappa, shape, dtype, 100)
+
+#@partial(jit, static_argnums = (3, 4, 5), inline = True)
+def _vonmises(key, x, kappa, shape, dtype, max_iters):
+
```

Comment:
For the type issue, add 
```suggestion
  x = lax.convert_element_type(x, dtype)
  kappa = lax.convert_element_type(kappa, dtype)
```

Reply:
Worked like a charm. Think I've found the other issues, shall update.